### DISCUSSION TITLE
Get geometry of object in the scene for ray tracing

### DISCUSSION THREAD

**Discussion by jmoraispk** [NVlabs/sionna#198] [Q&A] [answered]
After importing a scene with multiple buildings, is it possible to obtain the geometry of each building? 

The final objective is to detect if a certain position is inside or outside a building.

**Comment by faycalaa** [comment, marked as answer]
Hi,

Sionna does not currently offer a feature that determines whether a given position is inside or outside an object.

However, you can retrieve the geometry of the objects in the scenes by accessing the relevant [Mitsuba Shape](https://mitsuba.readthedocs.io/en/stable/src/api_reference.html#mitsuba.Shape).

Here's an example:

```
import sionna
# Import mitsuba and drjit
import mitsuba as mi
import drjit as dr

# Load a scene
scene = sionna.rt.load_scene(sionna.rt.scene.floor_wall)
# Get the Mitsuba scene
mi_scene = scene.mi_scene

# Loop through the Mitsuba shapes and get the one named 'floor' (as an example)
for shape in mi_scene.shapes():
    if shape.id() == 'mesh-floor': # A prefix 'mesh-' is added in the name (id) of the Mitsuba shapes
        break
mi_floor = shape

# Retrieve the geometry of the object.
# In the following code, that means the positions of the vertices,
# and the faces.
params = mi.traverse(mi_floor)
positions = params['vertex_positions']
positions = dr.unravel(mi.Point3f, positions)
faces = params['faces']
faces = dr.unravel(mi.Point3u, faces)

print(faces)
print(positions)
```

The "floor" object consists of a rectangle, which is composed of two triangles (primitives).
There are 4 vertices in total. Their coordinates can be found in the 4x3 ``positions`` arrays. It's important to note that multiple primitives can share the same vertices.
To identify which vertices form each triangle, you can refer to the 2x3 ``faces`` arrays. This arrays provides the indices of the vertices as they appear in the ``positions`` arrays.

Some useful links:
- [Mitsuba API documentation](https://mitsuba.readthedocs.io/en/stable/src/api_reference.html)
- [Dr.Jit API documentation](https://drjit.readthedocs.io/en/latest/reference.html)
